src/Cooked/Pretty/MockChain.hs

```diff
@@ -0,0 +1,210 @@
+      . map (\(addr, plSet) -> prettyItemize opts (prettyCookedOpt opts addr) "-" plSet)
+      . List.sortBy addressOrdering
+      . Map.toList
+      . utxoState
+    where
+      addressOrdering :: (Api.Address, a) -> (Api.Address, a) -> Ordering
+      addressOrdering
+        (a1@(Api.Address (Api.PubKeyCredential pkh1) _), _)
+        (a2@(Api.Address (Api.PubKeyCredential pkh2) _), _) =
+          case (walletPKHashToId pkh1, walletPKHashToId pkh2) of
+            (Just i, Just j) -> compare i j
+            (Just _, Nothing) -> LT
+            (Nothing, Just _) -> GT
+            (Nothing, Nothing) -> compare a1 a2
+      addressOrdering
+        (Api.Address (Api.PubKeyCredential _) _, _)
+        (Api.Address (Api.ScriptCredential _) _, _) = LT
+      addressOrdering (a1, _) (a2, _) = compare a1 a2
+
+-- | Pretty prints the state of an address, that is the list of UTxOs (including
+-- value and datum), grouped
+instance PrettyCookedL UtxoPayloadSet where
+  prettyCookedOptLM opts =
+    (prettyPayloadGrouped <$>)
+      . group
+      . List.sortBy (compare `on` (Api.lovelaceValueOf . utxoPayloadValue))
+      . utxoPayloadSet
+    where
+      similar :: UtxoPayload -> UtxoPayload -> Bool
+      similar
+        (UtxoPayload _ value1 skelOutDatum1 refScript1)
+        (UtxoPayload _ value2 skelOutDatum2 refScript2) =
+          value1 == value2
+            && skelOutDatum1 == skelOutDatum2
+            && refScript1 == refScript2
+
+      group :: [UtxoPayload] -> [[UtxoPayload]]
+      group =
+        case pcOptPrintTxOutRefs opts of
+          PCOptTxOutRefsFull -> map (: [])
+          _ -> List.groupBy similar
+
+      -- Pretty prints payloads (datum and value corresponding to 1 UTxO)
+      -- grouped together when they carry same value and datum
+      prettyPayloadGrouped :: [UtxoPayload] -> Maybe DocCooked
+      prettyPayloadGrouped [] = Nothing
+      prettyPayloadGrouped [payload] = prettyPayload (pcOptPrintTxOutRefs opts /= PCOptTxOutRefsHidden) payload
+      prettyPayloadGrouped (payload : rest) =
+        (PP.parens ("×" <> prettyCookedOpt opts (1 + length rest)) <+>)
+          <$> prettyPayload False payload
+
+      -- Optionally prints a 'UtxoPayload' with an option piloting whether
+      -- 'Api.TxOutRef's should be shown.
+      prettyPayload :: Bool -> UtxoPayload -> Maybe DocCooked
+      prettyPayload showTxOutRef UtxoPayload {..} =
+        case catMaybes
+          [ if showTxOutRef
+              then Just $ prettyCookedOpt opts utxoPayloadTxOutRef
+              else Nothing,
+            Just (prettyCookedOpt opts utxoPayloadValue),
+            prettyCookedOptM opts utxoPayloadSkelOutDatum,
+            ("Reference script hash:" <+>) . prettyHash opts <$> utxoPayloadReferenceScript
+          ] of
+          [] -> Nothing
+          [doc] -> Just $ PP.align doc
+          docs -> Just . PP.align . PP.vsep $ docs
+
+newtype CollateralInput = CollateralInput {unCollateralInput :: Api.TxOutRef}
```

Comment:
I don't think this is a relevant module to define new types. This is supposed to be a library of PrettyCooked instance implementations

Reply:
Sadly however, this is necessary, because there isnt a single way to define a `PrettyCooked` instance for `TxOutRef`. Printing it depends on how this utxo is used. Wrapping it in those types allows to write different instances for each of them and thus have multiple instances for `TxOutRef`. These types are just wrappers and are not needed anywhere else in cooked. In particular, they are not exported by the modules.